### ISSUE TITLE
Versions after 10.1.0 give module error for existing tests

### ISSUE THREAD

**Issue by DriftBP** [getsaf/shallow-render#185]
This basic test has been in place for a while but now throws an error if I upgrade shallow-render to a version after 10.1.0.

```
Error:
 FAIL  src/app/profile/mixcloud-widget/mixcloud-widget.component.spec.ts (9.021 s)
  ● MixcloudWidgetComponent › should create

    Component ProfilesComponent is not part of any NgModule or the module has not been imported into your module.
```

ProfilesComponent is another component declared within the same module as the component being tested, but neither use each other. I have the same error with other components in other modules. I've tried every version up to and including 10.1.4 and get the same result.

The test:
```
import { Shallow } from 'shallow-render';

import { MixcloudWidgetComponent } from './mixcloud-widget.component';
import { ProfileModule } from '../profile.module';

describe('MixcloudWidgetComponent', () => {
  let shallow: Shallow<MixcloudWidgetComponent>;

  beforeEach(() => {
    shallow = new Shallow(MixcloudWidgetComponent, ProfileModule);
  });

  it('should create', async () => {
    const { element } = await shallow.render();

    expect(element.nativeElement).toBeTruthy();
  });
});

```

**Comment by getsaf** [comment]
Thanks, for the details, someone else mentioned this in another post recently too. I will look into it. First thing I need to do is to reproduce the issue so I'll try to mirror your module setup to see if I can get the same error.

**Comment by getsaf** [comment]
Does [this stackblitz](https://stackblitz.com/edit/github-wo9m7p?file=examples%2Fmultiple-components-one-unused.spec.ts) properly mimic your module setup? Is there anything special going on besides standard imports/exports like use of entryComponents or lazy loading or moduleWithProviders?

**Comment by getsaf** [comment]
Also, did you find that 10.0.x worked without this issue? Only after 10.1.x?

**Comment by DriftBP** [comment]
> Also, did you find that 10.0.x worked without this issue? Only after 10.1.x?

After 10.1.0 is the first time that I've had the issue. I update fairly regular so probably used a few different versions before this but didn't have an issue.

**Comment by DriftBP** [comment]
> Does [this stackblitz](https://stackblitz.com/edit/github-wo9m7p?file=examples%2Fmultiple-components-one-unused.spec.ts) properly mimic your module setup? Is there anything special going on besides standard imports/exports like use of entryComponents or lazy loading or moduleWithProviders?

The module is lazy loaded, yes, and I'm using Angular 10.

The module itself is pretty basic:
```
@NgModule({
  imports: [
    CommonModule,
    RouterModule,
    ProfileRoutingModule,
    SharedModule,
    ScheduleModule
  ],
  declarations: [
    ProfilesComponent,
    HostDetailsComponent,
    MixcloudWidgetComponent,
    TwitterWidgetComponent,
    ProfileButtonComponent
  ],
  providers: [
    ProfilesResolve,
    HostDetailsResolve,
    ProfileService
  ]
})
export class ProfileModule { }
```

**Comment by getsaf** [comment]
Thanks, I'm browsing the changes since version 10.0.0:
https://github.com/getsaf/shallow-render/compare/v10.0.0...master

Nothing jumps out at me that indicates a changes the way components in modules are handled except for some new provider-mocking logic but I will continue to look. Might be tomorrow before I figure it out. If you happen see anything in that diff that looks weird, let me know.

**Comment by getsaf** [comment]
I've tried reproducing this error in many ways but can't get it to happen on my sample NG10 project.

https://github.com/getsaf/shallow-render-ng10

I've tried:
* Creating a component and leaving it out of any module
* Using a barrel (index.ts) to re-export and use different imports in different places
* Lazy loading a module
* With/Without routing

Not sure how to get this to happen in a test. If someone can help reproduce this issue I'm sure can figure out a fix.

@Exomnius, you had this issue too, is there anything specific you can think of that could help me reproduce the issue?

**Comment by DriftBP** [comment]
@getsaf I do have my full repo available if you wanted to check it that way? https://github.com/DriftBP/breakpirates-v3

Run `npm run test` to see all's fine, then try an update of shallow-render to latest.

**Comment by getsaf** [comment]
Thanks, this helped me find the issue quickly.

It turns out that the AngularRoutingModule uses an InjectionToken to supply route info and a recent change in Shallow allowed useValue-InjectionTokens to be passed through without mocking since InjectionTokens are typically just a way to inject configuration into your application. It seems that using Jest to to run tests is somehow different from the way Karma runs tests and when the routing configuration is injected, the Angular Compiler notices that there is a component in the routes that is not found in the current test module tree.

Not sure why Jest triggers this but Karma does not but I think my fix is to just stop passing trough InjectionToken values directly and mock them instead.

I'll let you know how it turns out.

**Comment by getsaf** [comment]
Got it fixed in [v10.2.0](https://github.com/getsaf/shallow-render/releases/tag/v10.2.0), I just tried it out on your repo and it it's all passing now!

**Comment by DriftBP** [comment]
@getsaf Nice! I'll give it a try this evening 👍
